pkg/auth/auth.go

```diff
@@ -73,13 +73,17 @@ func PersistLogout(config *config.Config) error {
 
 func PersistConfluentLoginToConfig(cfg *config.Config, credentials *Credentials, url, token, caCertPath string, isLegacyContext, save bool) error {
 	if credentials.IsSSO {
-		sub, err := getSsoSub(token)
+		subClaim, err := jwt.GetClaim(token, "sub")
 		if err != nil {
 			return err
 		}
-		if sub != "" {
-			credentials.Username = sub
+
+		sub, ok := subClaim.(string)
```

Comment:
Isn't every claim a string? I think we can move this line inside of `GetClaim()`

Reply:
Nope, `exp` is a float64. That's why I put the responsibility for the type assertion outside of the GetClaim function.